aptos-node/src/lib.rs

```diff
@@ -689,24 +689,32 @@ pub fn setup_environment_and_start_node(
         _ => None,
     };
 
-    let jwk_consensus_runtime = if let Some(obj) = jwk_consensus_network_interfaces {
-        let ApplicationNetworkInterfaces {
-            network_client,
-            network_service_events,
-        } = obj;
-        let (reconfig_events, onchain_jwk_updated_events) = jwk_consensus_subscriptions.expect(
-            "JWK consensus needs to listen to NewEpochEvents and OnChainJWKMapUpdated events.",
-        );
-        let jwk_consensus_runtime = start_jwk_consensus_runtime(
-            network_client,
-            network_service_events,
-            vtxn_pool.clone(),
-            reconfig_events,
-            onchain_jwk_updated_events,
-        );
-        Some(jwk_consensus_runtime)
-    } else {
-        None
+    let maybe_jwk_consensus_key =
+        load_consensus_key_from_secure_storage(&node_config.consensus.safety_rules);
+    debug!("maybe_jwk_consensus_key={:?}", maybe_jwk_consensus_key);
+
+    let jwk_consensus_runtime = match (jwk_consensus_network_interfaces, maybe_jwk_consensus_key) {
+        (Some(interfaces), Ok(consensus_key)) => {
+            let ApplicationNetworkInterfaces {
+                network_client,
+                network_service_events,
+            } = interfaces;
+            let (reconfig_events, onchain_jwk_updated_events) = jwk_consensus_subscriptions.expect(
+                "JWK consensus needs to listen to NewEpochEvents and OnChainJWKMapUpdated events.",
+            );
+            let my_addr = node_config.validator_network.as_ref().unwrap().peer_id();
+            let jwk_consensus_runtime = start_jwk_consensus_runtime(
```

Comment:
We don't check feature flag here?

Reply:
it seems in the existing pattern,  it's epoch manager's job.